packages/core/src/contrib/Functions.ts

```diff
@@ -357,20 +362,86 @@ export const compDict = {
       contents: [markStart, markEnd].map(toPt),
     };
   },
-
+  /**
+   * Return series of elements that can render an arc SVG. See: https://css-tricks.com/svg-path-syntax-illustrated-guide/ for the "A" spec.
+   * @param pathType: either "open" or "closed." whether the SVG should automatically draw a line between the final point and the start point
+   * @param start: coordinate to start drawing the arc
+   * @param end: coordinate to finish drawing the arc
+   * @param radius: width and height of the ellipse to draw the arc along (i.e. [width, height])
+   * @param rotation: angle in degrees to rotate ellipse about its center
+   * @param largeArc: 0 to draw shorter of 2 arcs, 1 to draw longer
+   * @param arcSweep: 0 to rotate CCW, 1 to rotate CW
+   * @returns: Elements that can be passed to Path shape spec to render an SVG arc
+   */
+  arc: (
+    pathType: string,
+    start: Pt2,
+    end: Pt2,
+    radius: Pt2,
+    rotation: IVarAD,
+    largeArc: IVarAD,
+    arcSweep: IVarAD
+  ): IPathDataV<VarAD> => {
+    const pathTypeStr = pathType === "closed" ? "Closed" : "Open";
+    const st: IPt<IVarAD> = { tag: "Pt", contents: start };
+    const arc: IArc<IVarAD> = {
+      tag: "Arc",
+      contents: [radius, [rotation, largeArc, arcSweep], end],
+    };
+    const elems: Elem<VarAD>[] = [st, arc];
+    return {
+      tag: "PathDataV",
+      contents: [
+        {
+          tag: pathTypeStr,
+          contents: elems,
+        },
+      ],
+    };
+  },
+  /**
+   * Find intersection between a circle centered at [x1, y1] with radius r and a line segment between [x1, y1] and [x2, y2].
+   * @param x1, y1: centerpoint of circle, one endpoint of line segment
+   * @param x2, y2: endpoint of line segment
+   * @param r: radius of circle
+   * @returns: vector representation of the point of intersection
+   */
+  angleMarker: (
```

Comment:
My interpretation is this function computes a point along a line segment `[(x1, y1), (x2, y2)]` that is `r` away from `(x1, y1)`? If so, you can actually find the unit vector of the line segment and use `ops.vmove(start, r, dir)` to find the point.